nms/packages/magmalte/app/views/subscriber/SubscriberTable.js

```diff
@@ -161,8 +172,172 @@ async function exportSubscribers(props: ExportProps) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [open, setOpen] = React.useState(false);
+  const [error, setError] = React.useState('');
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+  const successCountRef = useRef(0);
+  const handleClick = event => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  /**
+   * Save subscribers
+   *
+   * @param {Array<SubscriberInfo>} subscribers Array of subscribers to Add, Edit or Delete.
+   * @param {boolean} edit Update existing subscriber(s).
+   */
+  const saveSubscribers = async (
+    subscribers: Array<SubscriberInfo>,
+    edit: boolean,
+  ) => {
+    const subscriberErrors = '';
+    for (
+      let i = 0, j = subscribers.length;
+      i < j;
+      i += SUBSCRIBERS_CHUNK_SIZE
+    ) {
+      const subscriberChunk: Array<mutable_subscriber> = subscribers
+        .slice(i, i + SUBSCRIBERS_CHUNK_SIZE)
+        .map(subscriber => {
+          const authKey =
+            subscriber.authKey && isValidHex(subscriber.authKey)
+              ? hexToBase64(subscriber.authKey)
+              : '';
+
+          const authOpc =
+            subscriber.authOpc !== undefined && isValidHex(subscriber.authOpc)
+              ? hexToBase64(subscriber.authOpc)
+              : '';
+          const newSubscriber: mutable_subscriber = {
+            active_apns: subscriber.apns,
+            active_policies: subscriber.policies,
+            id: subscriber.imsi,
+            name: subscriber.name,
+            lte: {
+              auth_algo: 'MILENAGE',
+              auth_key: authKey,
+              auth_opc: authOpc,
+              state: subscriber.state,
+              sub_profile: subscriber.dataPlan,
+            },
+          };
+          return newSubscriber;
+        }, []);
+      try {
+        // PUT subscribers
+        if (edit) {
+          subscriberChunk.map(subscriber => {
+            ctx.setState?.(subscriber.id, subscriber);
+          });
+        } else {
+          // POST subscriber chunk if no errors
+          if (subscriberErrors.length > 0) {
+            setError(subscriberErrors);
+            return false;
+          }
+          const success = await bulkAdd(subscriberChunk);
+          if (success) {
+            successCountRef.current =
+              successCountRef.current + subscriberChunk.length;
+          } else {
+            enqueueSnackbar('Saving subscribers failed', {
+              variant: 'error',
+            });
+            return;
+          }
+        }
+      } catch (e) {
+        const errMsg = e.response?.data?.message ?? e.message ?? e;
+        enqueueSnackbar('Saving subscribers failed : ' + errMsg, {
+          variant: 'error',
+        });
+      }
+    }
+    enqueueSnackbar(
+      ` Subscriber${successCountRef.current > 1 ? 's' : ''} saved successfully`,
```

Comment:
I don't think we're gaining much here with the added complexity. Let's keep things simple where possible.
```suggestion
      'Subscriber(s) saved successfully',
```